### ISSUE TITLE
NCD - Labs - Lipid Panel [6h]

### ISSUE THREAD

**Issue by jeanette-pelizzon** [TIP-Global-Health/eheza-app#550]
## Issue 

We are adding in a lab called Lipid Panel into the NCD labs section.

(1) Icon is found in the components file [here](https://www.figma.com/file/4Rbvi4x1Cxmwzq73Kwsqxk/Components?node-id=201%3A28&t=CpDodF28GKeXQkOh-0).

(2) The order screen for this lab looks and operates the same as most of the other labs. 
<img width="317" alt="Screen Shot 2022-11-16 at 11 49 17 AM" src="https://user-images.githubusercontent.com/77505464/202281691-be569ffc-2cde-408d-ab38-33e92470a817.png">

(3) The results screen for this lab operates differently than other labs. We have had to add in a question asking what format the results were returned in - mmol/L or mg/dL. This is because every lab has a machine that runs this test and the calibration of those machines varies across the different health centers and labs. This is a requirement so we need to figure out how to do this and report the results accurately on the progress report. 

Here is the design. Notes are in red. 
<img width="264" alt="Screen Shot 2022-11-16 at 11 31 14 AM" src="https://user-images.githubusercontent.com/77505464/202282481-0459c2e1-ddae-4b96-960d-061fd11631bc.png">

(4) Results from this lab should be populated on the progress report in the labs section.  Here are the normal ranges for results. 
<img width="373" alt="Screen Shot 2022-11-16 at 11 44 00 AM" src="https://user-images.githubusercontent.com/77505464/202282576-6332234a-c012-4754-b379-d64499b4a168.png">

**Comment by anvmn** [comment]
@jeanette-pelizzon 

> This is a requirement so we need to figure out how to do this and report the results accurately on the progress report.

What about definitions for this part?

Also, we need a design for how this should look like on progress report (since we got 4 values for single lab).

**Comment by jeanette-pelizzon** [comment]
Design: 

You will see that the lipid panel comes up with results as abnormal/normal as an overview. These should show up abnormal if any of the individual subtests are abnormal which would then tell the nurse to click through to view the results. 

<img width="442" alt="Screen Shot 2022-12-12 at 10 46 44 AM" src="https://user-images.githubusercontent.com/77505464/207129132-b42bb79b-4b76-4865-8525-a33c649db339.png">

Once they click into results this is what it will look like: 
<img width="399" alt="Screen Shot 2022-12-12 at 10 51 02 AM" src="https://user-images.githubusercontent.com/77505464/207129621-143d65bb-c23e-4b77-9b91-1482b66dfc83.png">

**Comment by adamhstewart** [comment]
@anvmn regarding the mg/dl and mmol/L conundrum, here is a solution:

- The interface we have for collecting the data is good as we don't know how the test results can be received.
- However, we should always store the value as mg/dl (which is the international standard).
- To do this we will need to do a conversion on the fly, which is simply: ` mg/dl = 18 × mmol/l`
- We can then eliminate the need to have 2 scales of normality.

cc @jeanette-pelizzon

**Comment by anvmn** [comment]
@adamhstewart 

In this case, I'd expect normal ranges to have same proportions.
However, this does not allign in any way. The relation is between 37 to 79 mg/dL for 1 mmol/L.
It differs even for same measurement min and max (see LDL for example).
![Selection_1399](https://user-images.githubusercontent.com/17882245/207271908-fc8fdd87-4b82-4b95-87af-d4b67b38bb44.png)


Online, I found that there's a conversion calculator.
Note that relation for Cholesterol and Triglycerides is not the same.

![Selection_1400](https://user-images.githubusercontent.com/17882245/207273721-0a7e8215-5f1d-4a55-bf11-db04009787c6.png)


Also, about storing values.
On backend, we have the value and the unit. These 2 are inputs from the form.
We need the unit on backend, so that we can show the form with it's original selection.
Now, if we need to store value in mg/dL, this may cause some confusion.
So, the question is if we really need it stored in mg/dL, or we can simply have it presented on progress report in mg/dL?

//cc @jeanette-pelizzon

**Comment by adamhstewart** [comment]
@anvmn Thanks for checking teh work - it appears I came up with a general formula, but there is a different conversion for cholesterol measurements (not sure why, but alas):

```
For Total Cholesterol, HDL cholesterol and LDL cholesterol:

- To get from mmol/L to mg/dL multiply by 38.66976;
- To get from mg/dL to mmol/L multiply by 0.02586.

For Triglycerides:

- To get from mmol/L to mg/dL multiply by 88.57396;
- To get from mg/dL to mmol/L, multiply by 0.01129.
```

https://www.mdapp.co/cholesterol-conversion-calculator-600/

> So, the question is if we really need it stored in mg/dL, or we can simply have it presented on progress report in mg/dL?

I think if we can effectively convert, it's important to store in a consistent unit - that way we can do meaningful data analysis.

Storing the original measurement unit is good though, for archival purposes.

**Comment by LudmilaGoichman** [comment]
@adamhstewart @anvmn 

Please help. I am confuse with the ranges.
1. the app has an error in the normal range for total cholesterol in mg/dL. The normal value is greater than 200

<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210392909-080ed862-0d65-4b2d-bc3c-810f99bb0bdc.png">

2. If the value of triglycerides in mmol/L is 0.67 and is below the normal range, then why is the value converted to mg/dL - 59.34 a normal value?

<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210394785-310d05ea-86fd-4292-8253-ac36349b3072.png">|<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210394715-12fe456f-1e8f-4e74-86b6-5b67c268b7c3.png">
---|---

3. I didn't know about the following formula and entered the wrong values. Maybe LCD or HDL or Total Cholesterol should be calculated?

https://www.omnicalculator.com/health/ldl

![Pasted_Image_03_01_2023__18_13](https://user-images.githubusercontent.com/35059802/210396529-58dbf749-7b7d-4850-9f41-3beded2cbcd5.png)
<img width="456" alt="LDL_Cholesterol_Calculator" src="https://user-images.githubusercontent.com/35059802/210397099-0ee021c6-87ed-437e-9562-d2a94b199a10.png">

4. should fasting or non-fasting be considered? ^^

**Comment by anvmn** [comment]
@LudmilaGoichman 

> 1. The app has an error in the normal range for total cholesterol in mg/dL. The normal value is greater than 200.

Requirements should not have used `<`. Total cholesterol should be bellow 200.
![Selection_1463](https://user-images.githubusercontent.com/17882245/210421516-f81e8d0c-bd26-4a42-bad9-3c4bee7358d5.png)

> 2. If the value of triglycerides in mmol/L is 0.67 and is below the normal range, then why is the value converted to mg/dL - 59.34 a normal value
 
Do not bother with conversions. We only use mg/dL ranges. Those are the ones that should be tested.

> 3. I didn't know about the following formula and entered the wrong values. Maybe LCD or HDL or Total Cholesterol should be calculated?

I believe that calculations are performed at the lab. We only input those results.

> 4. should fasting or non-fasting be considered? ^^

Right now, there were no such requirements.
@jeanette-pelizzon do you know anything about it?

**Comment by anvmn** [comment]
@LudmilaGoichman with the answers above, I do not see a problem to proceed with QA for this issue, so I assign it back to you.

**Comment by LudmilaGoichman** [comment]
@anvmn @jeanette-pelizzon 

- Nutrition env.

**I only checked the ranges in mg/dL**
> Do not bother with conversions. We only use mg/dL ranges. Those are the ones that should be tested.

**The ranges in mg/dL:**
TC:    < 200
LDL:  130 - 160
HGL: 40 - 60
TG:    54 - 150

<img width="793" alt="NCD_-_Labs_-_Lipid_Panel__6h__·_Issue__550_·_TIP-Global-Health_eheza-app" src="https://user-images.githubusercontent.com/35059802/210526377-0f6bd367-9014-408a-abb9-44d32b05f25c.png">

### Questions:
1. The issue 4 from https://github.com/TIP-Global-Health/eheza-app/issues/417#issuecomment-1369135701 fixed in https://github.com/TIP-Global-Health/eheza-app/pull/581 , but in this fixed bug the entry in case management disappears after refreshing the page, and when testing the Lipid Panel, the entries don't disappear at all. I hope this is also fixed in https://github.com/TIP-Global-Health/eheza-app/pull/581

<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210558483-628c30fd-2861-46e2-bca9-c10b49757cf4.png">

2. This test doesn't exist in Antenatal Care encounter, but the report shown at https://github.com/TIP-Global-Health/eheza-app/issues/550#issuecomment-1347077548 is from Antenatal Care. Should a test be added to this encounter?

<img width="490" alt="207129132-b42bb79b-4b76-4865-8525-a33c649db339_png__884×1244_" src="https://user-images.githubusercontent.com/35059802/210565094-7ed8f7ac-a9fc-4ef5-8f89-e8f8ab598914.png">

3. This question is not related to this issue, but I cannot be silent. Sorry.

Antenatal Care encounter. this screen looks bad and annoying. The space is missing under the word "Birth":

<img width="361" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210565900-3de94e05-4adc-4f83-84de-4293b655cd14.png">


### Checked and Approved:
1. The icon is correct:

active | not active |
---|---
<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210531356-1a426f9e-feba-4091-82c4-b74f564d77c7.png">|<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210531563-d7f26dbd-fd12-45fd-9afc-071472bdb74e.png">


5. The flow (screens) is correct. Example:

Activity:

<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210531400-eb63b7b7-7b4f-4167-8667-04c16b74d781.png">|<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210531451-9b714c9c-5b9d-459c-8690-16a70b429024.png">|<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210531737-c55d0b78-977b-42d1-a492-7325ec3e5bd1.png">|<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210531802-c20a1187-333f-4e62-b3f7-8216b480907b.png">|<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210531889-fd0c8043-e730-4400-bb56-9b9319fb0851.png">
---|---|---|---|---

Lab Results:

<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210532118-63caab56-4a31-4dd5-b8dd-b4dc061bf294.png">|<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210532495-0a62ac5b-b749-429c-9b30-e920357e7d42.png">
---|---

6. The lab results. The question asking what format the results were returned in - mmol/L or mg/dL has been added. The selected units are displayed next to the numeric fields.

<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210532249-3f805221-67d7-479b-91e0-f56143180e70.png">|<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210532317-fb93cbef-504c-4f16-9f43-e6d9705bd3d4.png">|<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210532423-c5713ddc-3f66-4da4-9ace-56779751461b.png">|<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210532576-00adff98-533e-417c-97c4-07f3f22132b1.png">
---|---|---|---

7. NCD. Report looks as expected (see images in the point 6)
8. NCD. A test row in the Lab Results Table is displayed as abnormal if any of the individual subtests is abnormal. 

at least one subtest is abnormal | all subtests are normal |
---|---
<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210536565-8a8abae9-dde6-46ef-96cd-8846a4ca08cf.png">|<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210536895-ed487a30-6209-4b42-8a53-6ec845c2cb4b.png">

9. NCD. Results ranges in mg/dL are correct:

<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210533816-712c4c6d-1b77-4785-90ac-a0c82939e506.png">|<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210533903-57227820-bbd5-49e0-af6d-7fa2be26c5d6.png">|<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210533991-5958c78a-0ead-4f90-b842-7d8800e0533b.png">
---|---|---

<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210536473-015b7055-b75d-4e2a-aaa5-562a1226fc7b.png">|<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210536541-4db99a32-2d76-4371-9196-eba375736fde.png">|<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210536671-a9cc44c1-486e-4518-b86c-020a8bd59e3d.png">
---|---|---

<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210536806-a9a6d54c-b4a9-45c6-927a-b0f58e10011d.png">|<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210536863-58a86649-6000-4b90-a891-4b8ccd0a59bd.png">|<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210537108-e8d52d87-3a50-4959-9449-c0774bb18e16.png">
---|---|---

<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210537228-9d93c58e-8ae2-44f8-8dc1-dc871e1307f4.png">|<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210537355-d05ec910-8504-48c0-b8e0-9cca620ae04e.png">|<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210537453-ba24d2a1-d431-41bc-8405-afc353be5383.png">
---|---|---

10. NCD. Subsequent encounter report:

<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210556597-ce98a89b-8997-409a-aa4b-1de221838ac6.png">|<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210556678-01833396-91bf-48c7-a4b9-ca7d98d11e37.png">|<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210556751-d2007fb8-ec44-4a5d-9633-4f66911304c3.png">
---|---|---

<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210556804-c89512d4-8a16-49ac-993c-2ae729fb4741.png">|<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210556860-061fb19e-292d-4616-bb76-5ab9c03c1cdd.png">|<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/210556926-52f85594-3e2d-411c-a1c9-45df81837a51.png">
---|---|---

**Comment by anvmn** [comment]
@LudmilaGoichman 

Points 1 and 3 fixed at #581.

> 2. This test doesn't exist in Antenatal Care encounter, but the report shown at ...

The design is not accurate here. The test appears only at NCD encounter.

**Comment by LudmilaGoichman** [comment]
@anvmn @jeanette-pelizzon 

- nutrition env

### Checked and Approved:
1. Case management entry disappears after entering test(s) results

<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/211199330-e1f4d7dc-910b-462c-971d-5f14dd4ad167.png">|<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/211199349-5dbc87ce-fe82-458b-aa21-6e34c27cb460.png">
---|---

3. Antenatal Care encounter. Added space under the word "Birth" in this screen:

<img width="365" alt="Screenshots_-_Google_Drive" src="https://user-images.githubusercontent.com/35059802/211199466-bf35af4a-3c61-4206-a72a-ecd4190bad45.png">
